BDArmory/Competition/RemoteOrchestration/BDAScoreClient.cs

```diff
@@ -483,33 +484,45 @@ private void SaveCraftFile(VesselModel vessel, byte[] bytes)
                 .ToArray();
             File.WriteAllLines(filename, modifiedLines);
             Debug.Log(string.Format("[BDArmory.BDAScoreClient] Saved craft for player {0}", vesselName));
-            if (vesselName.Contains(BDArmorySettings.REMOTE_ORCHESTRATION_NPC_SWAPPER)) //grab either ships or players that contain NPC identifier
-            {
-                SwapCraftFiles(vesselName); //doing this after initial load/editing to make sure nothing breaks by swapping earlier
-            }
+            //if (vesselName.Contains(BDArmorySettings.REMOTE_ORCHESTRATION_NPC_SWAPPER)) //grab either ships or players that contain NPC identifier
+            //{
+            //    SwapCraftFiles(vesselName); //doing this after initial load/editing to make sure nothing breaks by swapping earlier
+            //}
         }
 
-        public void SwapCraftFiles(string vesselname)
+        public void SwapCraftFiles()
         {
-            string filename = string.Format("{0}/{1}.craft", vesselPath, vesselname);
-
-            Debug.Log("[BDArmory.BDAScoreClient] Swapping existing craft in spawn directory " + vesselPath);
-            DirectoryInfo info = new DirectoryInfo(NPCPath);
+            DirectoryInfo info = new DirectoryInfo(vesselStagingPath);
             FileInfo[] craftFiles = info.GetFiles("*.craft")
                 .Where(e => e.Extension == ".craft")
                 .ToArray();
-            int i;
-            i = (int)UnityEngine.Random.Range(0, craftFiles.Count() - 1);
-
-            string NPCfilename = string.Format("{0}/{1}", NPCPath, craftFiles[i].Name); //.craft included in the craftFiles[i].name
-            string[] NPClines = File.ReadAllLines(NPCfilename); //kludge, probably easier to just copy the file from NPC dir to the autospawn dir
-            string pattern = ".*ship = (.+)";
-            string[] modifiedLines = NPClines
-                .Select(e => Regex.Replace(e, pattern, "ship = " + vesselname))
-                .Where(e => !e.Contains("VESSELNAMING"))
-                .ToArray();
-            File.WriteAllLines(filename, modifiedLines);
-            Debug.Log(string.Format("[BDArmory.BDAScoreClient] Swapped craft for player {0}", vesselname));
+            foreach (FileInfo file in craftFiles)
+            {
+                if (file.Name.Contains(BDArmorySettings.REMOTE_ORCHESTRATION_NPC_SWAPPER))
+                {
+                    string vesselname = file.Name;
+                    string filename = string.Format("{0}/{1}", vesselStagingPath, vesselname);
+                    vesselname = vesselname.Remove(vesselname.Length - 6, 6); //cull the .craft from the string
+                    Debug.Log("[BDArmory.BDAScoreClient] Swapping existing craft " + vesselname + " in spawn directory");
+                    DirectoryInfo NPCinfo = new DirectoryInfo(NPCPath);
+                    FileInfo[] NPCFiles = NPCinfo.GetFiles("*.craft")
+                        .Where(e => e.Extension == ".craft")
```

Comment:
The `.Where(...).ToArray()` is unnecessary since you're globbing for `"*.craft"` in `GetFiles`.
Also, this could be outside the `foreach` loop with `craftFiles` since it doesn't appear to be changing for each `file`.